src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -93,50 +172,239 @@ internal NativeIntegerTypeSymbol(NamedTypeSymbol underlyingType) : base(underlyi
+        {
+            // Emit should use underlying symbol only.
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        private sealed class NativeIntegerParameterSymbol : WrappedParameterSymbol, Cci.IReference
+        {
+            private readonly NativeIntegerTypeSymbol _containingType;
+            private readonly Symbol _container;
+
+            internal NativeIntegerParameterSymbol(NativeIntegerTypeSymbol containingType, Symbol container, ParameterSymbol underlyingParameter) : base(underlyingParameter)
+            {
+                _containingType = containingType;
+                _container = container;
+            }
+
+            public override Symbol ContainingSymbol => _container;
+
+            public override TypeWithAnnotations TypeWithAnnotations => _containingType.SubstituteUnderlyingType(UnderlyingParameter.TypeWithAnnotations);
+
+            void Cci.IReference.Dispatch(Cci.MetadataVisitor visitor)
+            {
+                // Emit should use underlying symbol only.
+                throw ExceptionUtilities.Unreachable;
             }
         }
     }
+
+    internal sealed class NativeIntegerPropertySymbol : WrappedPropertySymbol, Cci.IReference
+    {
+        private readonly NativeIntegerTypeSymbol _container;
+
+        internal NativeIntegerPropertySymbol(NativeIntegerTypeSymbol container, PropertySymbol underlyingProperty) : base(underlyingProperty)
+        {
+            Debug.Assert(underlyingProperty.ParameterCount == 0);
+            _container = container;
+            NativeIntegerTypeSymbol.VerifyEquality(this, underlyingProperty);
+        }
+
+        public override Symbol ContainingSymbol => _container;
+
+        public override TypeWithAnnotations TypeWithAnnotations => _container.SubstituteUnderlyingType(_underlyingProperty.TypeWithAnnotations);
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => UnderlyingProperty.RefCustomModifiers;
+
+        public override ImmutableArray<ParameterSymbol> Parameters => ImmutableArray<ParameterSymbol>.Empty;
+
+        public override MethodSymbol? GetMethod => MakeAccessor(UnderlyingProperty.GetMethod);
+
+        public override MethodSymbol? SetMethod => MakeAccessor(UnderlyingProperty.SetMethod);
+
+        public override ImmutableArray<PropertySymbol> ExplicitInterfaceImplementations =>
+            UnderlyingProperty.ExplicitInterfaceImplementations.SelectAsArray(
+                (Func<PropertySymbol, NativeIntegerTypeSymbol, PropertySymbol>)((property, map) => property.OriginalDefinition.AsMember(_container.SubstituteUnderlyingType(property.ContainingType))),
+                _container);
+
+        internal override bool MustCallMethodsDirectly => _underlyingProperty.MustCallMethodsDirectly;
+
+        private NativeIntegerMethodSymbol? MakeAccessor(MethodSymbol? accessor) => accessor is null ? null : new NativeIntegerMethodSymbol(_container, accessor, this);
```

Comment:
>new NativeIntegerMethodSymbol(_container, accessor, this) [](start = 109, length = 57)

This is unexpected we should return the instance already cached on the containing type.